csrc/scheduler/pointwise_utils.h

```diff
@@ -74,5 +30,44 @@ inline int64_t nRootDims(const TensorView* tv) {
   return tv_n_dims;
 }
 
+class DomainMap : public scheduler_tools::DomainMap {
+ public:
+  using scheduler_tools::DomainMap::DomainMap;
+
+  // The pointwise scheduler heuristics requires a minimum number of axes.
+  // The output reference tensor should respect this requirement.
+  TensorView* findReferenceTensorView(int64_t minimum_num_axes = 0) const {
+    TensorView* result = nullptr;
+    int64_t max_dims = -1;
+    for (auto output_tv :
+         ir_utils::filterByType<TensorView>(fusion_->outputs())) {
+      if (isValidReference(output_tv) &&
+          hasMinimumSize(output_tv, minimum_num_axes) &&
+          !output_tv->isFusionInput()) {
+        int64_t n_dims = pointwise_utils::nRootDims(output_tv);
+        if (n_dims > max_dims) {
+          result = output_tv;
+          max_dims = n_dims;
+        }
+      }
+    }
```

Comment:
Nit: I'd put long implementations like this to cpp for faster compilation. Otherwise, it would have to be compiled for each compilation unit including this header, which seems unnecessary.

Reply:
Yeah, agreed, the long compilation time is painful...